src/elements/Bookmark/Bookmark.tsx

```diff
@@ -1,124 +1,24 @@
-    const provider = showUrl ? url : oembed.provider_name || hostname(oembed.provider_url || url);
-
+export function Bookmark({ node }: BookmarkProps) {
     return (
-        <a
-            id={id}
-            className="prezly-slate-bookmark__provider"
-            rel="noopener noreferrer"
-            target="_blank"
-            href={providerUrl}
-        >
-            <img
-                className="prezly-slate-bookmark__provider-icon"
-                src={favicon}
-                alt={`${provider} favicon`}
-                aria-hidden="true"
-            />
-            <span className="prezly-slate-bookmark__provider-name">{provider}</span>
-        </a>
-    );
-};
-
-export const Bookmark: FunctionComponent<Props> = ({ node, className, ...attributes }) => {
-    const { url, oembed, layout, uuid } = node;
-    const showThumbnail = node.show_thumbnail && oembed.thumbnail_url;
-    const isEmpty = !showThumbnail && isEmptyText(oembed.title) && isEmptyText(oembed.description);
-    const actualLayout = showThumbnail ? layout : BookmarkCardLayout.HORIZONTAL;
-    const id = `bookmark-${uuid}`;
-
-    return (
-        <div
-            {...attributes}
-            className={classNames('prezly-slate-bookmark', {
-                'prezly-slate-bookmark--minimal': isEmpty,
-                'prezly-slate-bookmark--vertical': actualLayout === BookmarkCardLayout.VERTICAL,
-                'prezly-slate-bookmark--horizontal': actualLayout === BookmarkCardLayout.HORIZONTAL,
-            })}
-        >
-            {showThumbnail && oembed.thumbnail_url && (
-                <Thumbnail
-                    href={url}
-                    src={oembed.thumbnail_url}
-                    width={oembed.thumbnail_width}
-                    height={oembed.thumbnail_height}
-                />
-            )}
-            <div className="prezly-slate-bookmark__details">
-                {!isEmptyText(oembed.title) && (
-                    <a
-                        id={id}
-                        className="prezly-slate-bookmark__title"
-                        href={url}
-                        rel="noopener noreferrer"
-                        target="_blank"
-                    >
-                        {oembed.title}
-                    </a>
-                )}
-                {!isEmptyText(oembed.description) && (
-                    <div className="prezly-slate-bookmark__description">{oembed.description}</div>
-                )}
-                <Provider
-                    oembed={oembed}
-                    showUrl={isEmpty}
-                    id={isEmptyText(oembed.title) ? id : undefined}
-                />
-            </div>
-        </div>
+        <BookmarkCard
+            hrefId={`bookmark-${node.uuid}`}
+            node={{
+                layout: node.layout,
+                newTab: node.new_tab,
+                showThumbnail: node.show_thumbnail,
+                uuid: node.uuid,
+            }}
```

Comment:
Why don't we just unnest these to be top-level properties? IMO this is an unnecessary complication to use an object.

Reply:
imo in this case we at least can see to what type the fields belong and see clear name for them, rather that have random fields with type string, boolean etc.

Reply:
In my thinking, the `components` folder is a set of generic components, not bound to the concept of Slate nodes. In that sense, I see the `node` structure as an unnecessary leakage of concept to where it does not belong.